src/edu/ucsb/nceas/metacat/dataone/D1NodeService.java

```diff
@@ -2477,4 +2479,112 @@ public String getUserAgent() {
   public void setUserAgent(String userAgent) {
       this.userAgent = userAgent;
   }
+  
+  /**
+   * Compare two AccessPolicy objects
+   * @param ap1
+   * @param ap2
+   * @return true if they are same; false otherwise
+   */
+  public static boolean equals(AccessPolicy ap1, AccessPolicy ap2) {
+      boolean equal = false;
+      if (ap1 == null && ap2 == null) {
+          equal = true;
+      } else if (ap1 == null && ap2 != null) {
+          if (ap2.getAllowList() == null || ap2.getAllowList().isEmpty()) {
+              equal = true;
+          }
+      } else if (ap2 == null && ap1 != null) {
+          if (ap1.getAllowList() == null || ap1.getAllowList().isEmpty()) {
+              equal = true;
+          }
+      } else {
+          List<AccessRule> list1 = ap1.getAllowList();
+          List<AccessRule> list2 = ap2.getAllowList();
+          if (list1 == null && list2 == null) {
+              equal = true;
+          } else if (list1 == null && list2 != null) {
+              if (list2.isEmpty()) {
+                  equal = true;
+              }
+          } else if (list2 == null && list1 != null) {
+              if (list1.isEmpty()) {
+                  equal = true;
+              }
+          } else {
+              Map<String, List<Permission>> map1 = consolidateAccessRules(list1);
+              Map<String, List<Permission>> map2 = consolidateAccessRules(list2);
+              if (map1.size() == map2.size()) {
+                  outerLoop:
+                  for (String subject : map1.keySet()) {
+                      if (map2.get(subject) == null) {
+                          logMetacat.debug("D1NodeService.equals - found the subject " + subject + " does exist in the first AccessPolicy but doesnot exist in the second AccessPolicy. So they donot equal.");
```

Comment:
```suggestion
                          logMetacat.debug("D1NodeService.equals - found the subject " + 
                          subject +
                          " does exist in the first AccessPolicy but does not exist in the second AccessPolicy. " +
                          "So they are not equal.");
```
In general Jing, let's try to keep lines shorter than 100 characters so they don't wrap - it makes the code much more readable.